static/app/utils/discover/eventView.tsx

```diff
@@ -297,8 +297,19 @@ class EventView {
     const environment = Array.isArray(props.environment) ? props.environment : [];
 
     // only include sort keys that are included in the fields
+    let equations = 0;
     const sortKeys = fields
-      .map(field => getSortKeyFromField(field, undefined))
+      .map(field => {
+        if (isEquation(field.field)) {
+          const sortKey = getSortKeyFromField(
+            {field: `equation[${equations}]`},
+            undefined
+          );
+          equations += 1;
+          return sortKey;
+        }
+        return getSortKeyFromField(field, undefined);
+      })
```

Comment:
Don't forget to include a test for this.

Reply:
Thanks for the reminder, will be doing a pass later to add tests everywhere, its in my backlog :+1: